File: python/meteroid/api/batch_jobs.py
# this file is @generated
import typing as t

from ..models import (
    BatchJobChunkId,
    BatchJobDetailResponse,
    BatchJobFailuresResponse,
    BatchJobListResponse,
    BatchJobStatus,
    BatchJobType,
)
from .common import ApiBaseAsync, ApiBaseSync, decode_response, serialize_query_params


class BatchJobsAsync(ApiBaseAsync):
    """batch jobs API."""

    async def list_batch_jobs(
        self,
        *,
        job_type: t.Optional[BatchJobType] = None,
        status: t.Optional[t.List[BatchJobStatus]] = None,
        page: t.Optional[int] = None,
        per_page: t.Optional[int] = None,
    ) -> BatchJobListResponse:
        """List batch jobs with optional filtering by type and status.

        :param page: Page number (0-indexed)
        :param per_page: Number of items per page"""
        response = await self._request_asyncio(
            method="get",
            path="/api/v1/batch-jobs",
            query_params=serialize_query_params(
                {
                    "job_type": job_type,
                    "status": status,
                    "page": page,
                    "per_page": per_page,
                },
            ),
        )
        return decode_response(response, BatchJobListResponse)

    async def get_batch_job(
        self,
        batch_job_id: str,
    ) -> BatchJobDetailResponse:
        """Retrieve a single batch job with its chunks and failures."""
        response = await self._request_asyncio(
            method="get",
            path="/api/v1/batch-jobs/{batch_job_id}",
            path_params={
                "batch_job_id": batch_job_id,
            },
        )
        return decode_response(response, BatchJobDetailResponse)

    async def list_batch_job_failures(
        self,
        batch_job_id: str,
        *,
        chunk_id: t.Optional[BatchJobChunkId] = None,
        limit: t.Optional[int] = None,
        offset: t.Optional[int] = None,
    ) -> BatchJobFailuresResponse:
        """Retrieve paginated failures for a batch job."""
        response = await self._request_asyncio(
            method="get",
            path="/api/v1/batch-jobs/{batch_job_id}/failures",
            path_params={
                "batch_job_id": batch_job_id,
            },
            query_params=serialize_query_params(
                {
                    "chunk_id": chunk_id,
                    "limit": limit,
                    "offset": offset,
                },
            ),
        )
        return decode_response(response, BatchJobFailuresResponse)


class BatchJobs(ApiBaseSync):
    """batch jobs API."""

    def list_batch_jobs(
        self,
        *,
        job_type: t.Optional[BatchJobType] = None,
        status: t.Optional[t.List[BatchJobStatus]] = None,
        page: t.Optional[int] = None,
        per_page: t.Optional[int] = None,
    ) -> BatchJobListResponse:
        """List batch jobs with optional filtering by type and status.

        :param page: Page number (0-indexed)
        :param per_page: Number of items per page"""
        response = self._request_sync(
            method="get",
            path="/api/v1/batch-jobs",
            query_params=serialize_query_params(
                {
                    "job_type": job_type,
                    "status": status,
                    "page": page,
                    "per_page": per_page,
                },
            ),
        )
        return decode_response(response, BatchJobListResponse)

    def get_batch_job(
        self,
        batch_job_id: str,
    ) -> BatchJobDetailResponse:
        """Retrieve a single batch job with its chunks and failures."""
        response = self._request_sync(
            method="get",
            path="/api/v1/batch-jobs/{batch_job_id}",
            path_params={
                "batch_job_id": batch_job_id,
            },
        )
        return decode_response(response, BatchJobDetailResponse)

    def list_batch_job_failures(
        self,
        batch_job_id: str,
        *,
        chunk_id: t.Optional[BatchJobChunkId] = None,
        limit: t.Optional[int] = None,
        offset: t.Optional[int] = None,
    ) -> BatchJobFailuresResponse:
        """Retrieve paginated failures for a batch job."""
        response = self._request_sync(
            method="get",
            path="/api/v1/batch-jobs/{batch_job_id}/failures",
            path_params={
                "batch_job_id": batch_job_id,
            },
            query_params=serialize_query_params(
                {
                    "chunk_id": chunk_id,
                    "limit": limit,
                    "offset": offset,
                },
            ),
        )
        return decode_response(response, BatchJobFailuresResponse)

File: go/batch_jobs.go
// this file is @generated
package meteroid

import (
	"context"
	"net/http"
)

// BatchJobsListBatchJobsOptions carries the query and header parameters of
// BatchJobs.ListBatchJobs.
//
// Optional parameters are pointers; leave them nil to omit them.
type BatchJobsListBatchJobsOptions struct {
	JobType *BatchJobType

	Status []BatchJobStatus

	// Page number (0-indexed)
	Page *int32

	// Number of items per page
	PerPage *int32
}

// BatchJobsListBatchJobFailuresOptions carries the query and header parameters of
// BatchJobs.ListBatchJobFailures.
//
// Optional parameters are pointers; leave them nil to omit them.
type BatchJobsListBatchJobFailuresOptions struct {
	ChunkId *BatchJobChunkId

	Limit *int32

	Offset *int32
}

// BatchJobs groups the batch jobs operations of the Meteroid API.
type BatchJobs struct {
	client *Client
}

// List batch jobs with optional filtering by type and status.
func (a *BatchJobs) ListBatchJobs(ctx context.Context, options *BatchJobsListBatchJobsOptions) (*BatchJobListResponse, error) {
	req := newRequest(http.MethodGet, "/api/v1/batch-jobs")

	if options != nil {
		if options.JobType != nil {
			req.SetQueryParam("job_type", string(*options.JobType))
		}

		if len(options.Status) > 0 {
			for _, item := range options.Status {
				req.AddQueryParam("status", string(item))
			}
		}
		if options.Page != nil {
			req.SetQueryParam("page", formatInt(int64(*options.Page)))
		}
		if options.PerPage != nil {
			req.SetQueryParam("per_page", formatInt(int64(*options.PerPage)))
		}
	}

	var out BatchJobListResponse
	if err := a.client.execute(ctx, req, &out); err != nil {
		return nil, err
	}
	return &out, nil
}

// Retrieve a single batch job with its chunks and failures.
func (a *BatchJobs) GetBatchJob(ctx context.Context, batchJobId string) (*BatchJobDetailResponse, error) {
	req := newRequest(http.MethodGet, "/api/v1/batch-jobs/{batch_job_id}")
	req.SetPathParam("batch_job_id", batchJobId)

	var out BatchJobDetailResponse
	if err := a.client.execute(ctx, req, &out); err != nil {
		return nil, err
	}
	return &out, nil
}

// Retrieve paginated failures for a batch job.
func (a *BatchJobs) ListBatchJobFailures(ctx context.Context, batchJobId string, options *BatchJobsListBatchJobFailuresOptions) (*BatchJobFailuresResponse, error) {
	req := newRequest(http.MethodGet, "/api/v1/batch-jobs/{batch_job_id}/failures")
	req.SetPathParam("batch_job_id", batchJobId)

	if options != nil {
		if options.ChunkId != nil {
			req.SetQueryParam("chunk_id", string(*options.ChunkId))
		}
		if options.Limit != nil {
			req.SetQueryParam("limit", formatInt(int64(*options.Limit)))
		}
		if options.Offset != nil {
			req.SetQueryParam("offset", formatInt(int64(*options.Offset)))
		}
	}

	var out BatchJobFailuresResponse
	if err := a.client.execute(ctx, req, &out); err != nil {
		return nil, err
	}
	return &out, nil
}
